homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,447 @@
+        self.set_advanced_config(self.advanced_config)
+        self.set_schedule(self.schedule)
+
+    def set_advanced_config(self, advanced_config):
+        """Set the thermostat advanced config"""
+        try:
+            device = self.connect()
+            if device.auth():
+                device.set_advanced(advanced_config["loop_mode"],
+                                    advanced_config["sen"],
+                                    advanced_config["osv"],
+                                    advanced_config["dif"],
+                                    advanced_config["svh"],
+                                    advanced_config["svl"],
+                                    advanced_config["adj"],
+                                    advanced_config["fre"],
+                                    advanced_config["pon"])
+        except timeout:
+            _LOGGER.error("set_advanced_config timeout")
+
+    def set_schedule(self, schedule):
+        """Set the thermostat schedule"""
+        try:
+            device = self.connect()
+            if device.auth():
+                device.set_schedule(schedule[CONF_WEEKDAY],
+                                    schedule[CONF_WEEKEND])
+        except timeout:
+            _LOGGER.error("set_schedule timeout")
+
+    def power_on_off(self, power):
+        """power on/off thermostat"""
+        try:
+            device = self.connect()
+            if device.auth():
+                if str(power) == STATE_IDLE:
+                    device.set_power(POWER_OFF)
+                else:
+                    device.set_power(POWER_ON)
+        except timeout:
+            _LOGGER.error("power_on_off timeout")
+
+    def set_temperature(self, temperature):
+        """Set the thermostat target temperature"""
+        try:
+            device = self.connect()
+            if device.auth():
+                device.set_temp(float(temperature))
+        except timeout:
+            _LOGGER.error("set_temperature timeout")
+
+    def set_operation_mode(self, mode):
+        """Set the thermostat operation mode: [on, off, auto]"""
+        try:
+            device = self.connect()
+            if device.auth():
+                if mode == STATE_AUTO:
+                    device.set_power(POWER_ON)
+                    device.set_mode(AUTO, self.loop_mode)
+                elif mode == STATE_MANUAL:
+                    device.set_power(POWER_ON)
+                    device.set_mode(MANUAL, self.loop_mode)
+                elif mode == STATE_IDLE:
+                    device.set_mode(MANUAL, self.loop_mode)
+                    if self.freeze == 1:
+                        device.set_temp(float(12))
+                    else :
+                        device.set_temp(float(0))
+                    device.set_power(POWER_OFF)
+        except timeout:
+            _LOGGER.error("set_operation_mode timeout")
+
+    def read_status(self):
```

Comment:
This is usually something that is part of a standalone library. I think it's borderline, and maybe we require a separate standalone library here between home assistant and the device.